src/allmydata/frontends/magic_folder.py

```diff
@@ -676,6 +707,7 @@ def _maybe_upload(ign, now=None):
                 now = time.time()
             fp = self._get_filepath(relpath_u)
             pathinfo = get_pathinfo(unicode_from_filepath(fp))
+            item.size = pathinfo.size
```

Comment:
Setting this attribute from the outside strikes me as fishy.

It seems like both `UploadItem` and `DownloadItem` instances have a size passed to their initializer.  I suppose that size could be wrong if changes were made after the item was first noticed (at least in the `UploadItem` case).  Is that why the size gets updated here?

But another change could always be made after this update... It's always possible for the information to be out of date, right?

Reply:
Yes, all that is true. I was setting it here to be "more sure" the size was as close to the real one as possible. Shall I just add a comment, or remove this entirely..?

Reply:
Maybe that `get_pathinfo` should be delayed until someone actually wants the information?  That would be as close to the real size as you can get, I think.  Probably not for this PR.  So maybe file a ticket for that and remove this from here for now?

Reply:
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2904

Reply:
(Hmm, possible alternative reasoning: since we're getting the pathinfo there *anyway*, setting `.size` will ensure that the history reflects our notion of the file-size when we started processing/uploading the item)